src/sync/atomic/mod.rs

```diff
@@ -0,0 +1,187 @@
+//! correctness, the [Loom] crate provides support for reasoning about Acquire and Release orderings
+//! and partial support for Relaxed orderings.
+//!
+//! To disable the warning printed about this issue, set the `SHUTTLE_SILENCE_ORDERING_WARNING`
+//! environment variable to any value, or call [`silence_ordering_warning`].
+//!
+//! [Loom]: https://crates.io/crates/loom
+
+mod bool;
+mod int;
+mod ptr;
+
+pub use self::bool::AtomicBool;
+pub use int::*;
+pub use ptr::AtomicPtr;
+pub use std::sync::atomic::Ordering;
+
+use crate::runtime::thread;
+use std::cell::RefCell;
+
+static PRINTED_ORDERING_WARNING: std::sync::atomic::AtomicBool = std::sync::atomic::AtomicBool::new(false);
+
+#[inline]
+fn maybe_warn_about_ordering(order: Ordering) {
+    use ansi_term::Colour;
+
+    #[allow(clippy::clippy::collapsible_if)]
+    if order != Ordering::SeqCst {
+        if PRINTED_ORDERING_WARNING
+            .compare_exchange(false, true, Ordering::Relaxed, Ordering::Relaxed)
+            .is_ok()
+        {
+            if std::env::var("SHUTTLE_SILENCE_ORDERING_WARNING").is_ok() {
+                return;
+            }
+
+            eprintln!(
+                "{}: Shuttle only correctly models SeqCst atomics and treats all other Orderings \
+                as if they were SeqCst. Bugs caused by weaker orderings like {:?} may be missed. \
+                See https://docs.rs/shuttle/*/shuttle/sync/atomic/index.html#warning-about-relaxed-behaviors \
+                for details or to disable this warning.",
+                Colour::Yellow.normal().paint("WARNING"),
+                order
+            );
+        }
+    }
+}
+
+/// Disable the warning printed when Shuttle executes atomic operation with non-SeqCst orderings,
+/// which are not supported. See [the warning discussion](self#warning-about-relaxed-behaviors) for
+/// details.
+pub fn silence_ordering_warning() {
+    PRINTED_ORDERING_WARNING.store(true, Ordering::Relaxed);
+}
+
+/// An atomic fence, like the standard library's [std::sync::atomic::fence].
+pub fn fence(order: Ordering) {
+    if order == Ordering::Relaxed {
+        panic!("there is no such thing as a relaxed fence");
+    }
+
+    maybe_warn_about_ordering(order);
+
+    // SeqCst fences are no-ops in our execution model
+}
+
+// We can just reuse the standard library's compiler fence, as they have no visible run-time
+// behavior and so we need neither insert yieldpoints nor warn about non-SeqCst orderings.
+pub use std::sync::atomic::compiler_fence;
+
+/// Base type for atomic implementations. This type handles generating the right interleavings for
+/// all atomics. The interesting stuff is in `load`, `store`, `swap`, and `fetch_update`; all other
+/// atomic operations are implemented in terms of those four primitives.
+#[derive(Debug)]
+struct Atomic<T> {
+    inner: RefCell<T>,
+}
+
+unsafe impl<T: Sync> Sync for Atomic<T> {}
```

Comment:
Add a comment saying why this is okay for Shuttle (as we've done in other places)?